pkg/kubelet/cm/helpers_linux.go

```diff
@@ -202,8 +202,16 @@ func getCgroupSubsystemsV1() (*CgroupSubsystems, error) {
 	}
 	mountPoints := make(map[string]string, len(allCgroups))
 	for _, mount := range allCgroups {
+		// BEFORE kubelet used a random mount point per cgroups subsystem;
+		// NOW    more deterministic: kubelet use mount point with shortest path;
```

Comment:
Why shortest? Why not longest?

Reply:
Good question!

Here is an example from commit message:
```
Mountpoint: "/sys/fs/cgroup/systemd/some/path",
Mountpoint: "/sys/fs/cgroup/systemd",
```

In case of shortest a path most likely (comparing to random as was before) will be a root of cgroups fs.
It's not ideal solution since there is no guarantee of using the mount which we want since we don't know what we want : )
We just want to use some mount per subsystem.

Better choice would be to use strictly mounts with Root == "/". However tests with `kind` will never pass.

Reply:
@b10s I like what you proposed here strictly mounting with root == "/", which is much cleaner solution here.  Can you elaborate a little bit more why tests with kind will never pass?

Reply:
Never mind. I saw the failing tests below.

Reply:
@dchen1107 

```
 # For each cgroup subsystem, Docker does a bind mount from the current
  # cgroup to the root of the cgroup subsystem. For instance:
  #   /sys/fs/cgroup/memory/docker/<cid> -> /sys/fs/cgroup/memory
  #
  # This will confuse Kubelet and cadvisor and will dump the following error
  # messages in kubelet log:
  #   `summary_sys_containers.go:47] Failed to get system container stats for ".../kubelet.service"`
  #
  # This is because `/proc/<pid>/cgroup` is not affected by the bind mount.
  # The following is a workaround to recreate the original cgroup
  # environment by doing another bind mount for each subsystem.
```
  https://github.com/kubernetes-sigs/kind/blob/11b418d9882efbd7b2284495d0c737ab9130ae05/images/base/files/usr/local/bin/entrypoint#L105